crates/store/re_chunk/src/iter.rs

```diff
@@ -52,4 +53,171 @@ impl Chunk {
 
         itertools::izip!(data_times, row_ids, arrays)
     }
+
+    /// Returns the cell corresponding to the specified [`RowId`] for a given [`ComponentName`].
+    ///
+    /// This is `O(log(n))` if `self.is_sorted()`, and `O(n)` otherwise.
+    ///
+    /// Reminder: duplicated `RowId`s results in undefined behavior.
+    pub fn cell(
+        &self,
+        row_id: RowId,
+        component_name: &ComponentName,
+    ) -> Option<Box<dyn ArrowArray>> {
+        let list_array = self.components.get(component_name)?;
+
+        let row_id_128 = row_id.as_u128();
+        let row_id_time_ns = (row_id_128 >> 64) as u64;
+        let row_id_inc = (row_id_128 & (!0 >> 64)) as u64;
```

Comment:
nit: this is for `is_sorted() == true` only, right? So it could be in that branch of the  `if`

Also I assume you'll end up doing this sort of things in multiple place and you'll factor those binary shenanigan somewhere clean :)

Reply:
That's actually the first time ever we have to compare native Rust `RowId`s with arrow-encoded ones, yep.

We shall abstract once we reach the :sparkles: golden abstraction rule :sparkles:.